README.md

```diff
@@ -34,12 +37,81 @@ Installation
 ember install ember-sinon-qunit
 ```
 
-
 Usage
 ------------------------------------------------------------------------------
 
-Import Ember-Sinon-QUnit's `test` method into your tests in place of Ember-QUnit's test. This creates a Sinon `sandbox`
-around that test via Sinon's `test` API. Then, you can access the following Sinon functions via `this` within the test callback:
+To use, import the setup method into your `tests/test-helper.js` file and execute it.
+
+```js
+import setupSinon from 'ember-sinon-qunit';
+
+...
+
+setupSinon();
+```
+
+This will automatically wire-up `sinon`'s setup & restoration to QUnit `testStart` and `testDone` respectively.
+
+#### Accessing `sinon` from Within Tests
+
+In each test you are able to access `sinon` via the `sinon` object available as an import in your tests:
+
+```js
+import sinon from 'sinon';
+
+...
+
+test('very important test happening here', function(assert) {
+  const spy = sinon.spy();
+
+  ...
+});
+```
+
+The `sinon` object's state is automatically self-contained to each specific test, allowing you to 
+safely create mocks for your tests without worrying about any overrides leaking between each test.
+
+Migrating To `ember-sinon-qunit`
+------------------------------------------------------------------------------
+
+The above functionality replaces previous features within `ember-sinon-qunit`, 
+as well as the sister addons [`ember-sinon-sinoff`](https://github.com/scalvert/ember-sinon-sinoff) 
+and [`ember-sinon-sandbox`](https://github.com/scalvert/ember-sinon-sandbox). 
+Below, you will find simple instructions for migrating from each of these feature sets to the new patterns.
+
+### Migration from `sinon` 5+
+
+1. Import and consume `setupSinon`.
+1. Remove any manual calls to `sinon.restore()`. It won't hurt to leave them, but they are redundant now!
+
+### Migration from older versions of `sinon`
+
+1. Import and consume `setupSinon`.
+1. Remove calls to `sinon.createSandbox()`. Anywhere you used the `sandbox` object returned by this method, 
```

Comment:
I had a codemod to do this. I wonder if we should expand it/polish it up and advertise it here?

Reply:
Link?